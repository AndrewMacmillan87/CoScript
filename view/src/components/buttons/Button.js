import React from "react";
import PropTypes from "prop-types";

import "./buttons.css";

const Button = ({ onClick, id, classes, value }) => {
    return (
        <button onClick={onClick} id={id} className={classes} type='button'>
            <span className='btn-text'>
                <b>{value}</b>
            </span>
        </button>
    );
};

Button.propTypes = {
    onClick: PropTypes.func,
    classes: PropTypes.string,
    id: PropTypes.string,
    value: PropTypes.string
};

export default Button;
