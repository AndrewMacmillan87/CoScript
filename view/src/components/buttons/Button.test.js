import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import { reduxStore } from "../../redux/store/reduxStore";

import Button from "./Button";

describe("Button component", () => {
    test("renders without crashing", () => {
        global.scrollTo = jest.fn();
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <Button />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        global.scrollTo = jest.fn();
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <Button />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render a single button element with a className that matches the 'classes' prop", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Button
                        onClick={() => {}}
                        id='id'
                        classes='test-class'
                        value='Test'
                    />
                </Router>
            </Provider>
        );
        expect(wrapper.find("button").length).toEqual(1);
        expect(wrapper.find({ className: "test-class" }).length).toEqual(1);
    });
});
