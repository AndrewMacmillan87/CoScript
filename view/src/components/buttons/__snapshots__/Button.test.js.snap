// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<button
  type="button"
>
  <span
    className="btn-text"
  >
    <b />
  </span>
</button>
`;
